pkg/lib/aws/cloudwatch.go

```diff
@@ -64,3 +83,249 @@ func (c *Client) TagLogGroup(logGroup string, tagMap map[string]string) error {
+
+	// get current dashboard form cloudwatch
+	currDashboardOutput, err := c.CloudWatch().GetDashboard(&cloudwatch.GetDashboardInput{
+		DashboardName: aws.String(dashboardName),
+	})
+	if err != nil {
+		errors.WithStack(err)
+	}
+	// get body string from GetDashboard return object
+	currDashboardString := *currDashboardOutput.DashboardBody
+
+	//define interface to unmarshal received body string
+	var currDashboard CloudWatchDashboard
+	err = json.Unmarshal([]byte(currDashboardString), &currDashboard)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// get lowest element of cloudwatch. Needed to place metrics below all existing metrics
+	highestY, err := getHighestYDashboard(currDashboard)
+	if err != nil {
+		return err
+	}
+
+	// create widgets for title and metrics
+	apiTitleWidget := createTextWidget(1, highestY+1, 22, 1, "## API: "+nameAPI)
+	// top left widget
+	statCodeWidget := createMetricWidget(1, highestY+2, 11, 6, statCodeMetric(dashboardName, nameAPI), "Status Code", "Sum", 60, c.Region)
+	// top right widget
+	inFlightWidget := createMetricWidget(12, highestY+2, 11, 6, inFlightMetric(dashboardName, nameAPI), "In flight requests", "Sum", 10, c.Region)
+	// bottem left widget
+	latencyWidgetp50 := createMetricWidget(1, highestY+8, 11, 6, latencyMetric(dashboardName, nameAPI), "median request response time (60s)", "p50", 60, c.Region)
+	// bottom right widget
+	latencyWidgetp99 := createMetricWidget(12, highestY+8, 11, 6, latencyMetric(dashboardName, nameAPI), "p99 of request response time (60s)", "p99", 60, c.Region)
+
+	// append new API metrics widgets to existing widgets
+	currDashboard.Widgets = append(currDashboard.Widgets, apiTitleWidget, statCodeWidget, inFlightWidget, latencyWidgetp50, latencyWidgetp99)
+	currDashboardJSON, err := json.Marshal(currDashboard)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	// upload updated dashboard to cloudwatch
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(currDashboardJSON)),
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	return nil
+}
+
+// CreateDashboard creates a new dashboard (or clears an existing one if it already exists)
+func (c *Client) CreateDashboard(dashboardName string) error {
+
+	//create cloudwatch base body with title
+	cloudwatchBaseBody := CloudWatchDashboard{Start: "-PT1H", PeriodOverride: "inherit", Widgets: []CloudWatchWidget{createTextWidget(7, 0, 10, 1, "# CORTEX MONITORING DASHBOARD")}}
+	cloudwatchBaseBodyJSON, err := json.Marshal(cloudwatchBaseBody)
+	if err != nil {
+		return errors.Wrap(err, "failed to encode cloudwatch base body into json")
+	}
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(cloudwatchBaseBodyJSON)),
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to create dashboard: "+dashboardName)
```

Comment:
nit: `return errors.Wrap(err, "failed to create dashboard: "+dashboardName)` is equivalent to `return errors.Wrap(err, "failed to create dashboard", dashboardName)`. `errors.Wrap(...)`, delimits each item with `: `.

I would search for occurences of `: "+` and replace address them.